presto-hive/src/main/java/com/facebook/presto/hive/security/ranger/RangerBasedAccessControl.java

```diff
@@ -0,0 +1,516 @@
+    {
+        if (!checkAccess(identity, tableName, null, HiveAccessType.ALTER)) {
+            denyRenameTable(tableName.toString(), format("Access denied - User [ $s ] does not have is not have [ALTER] " +
+                    "privilege on [ %s/%s ] ", identity.getUser(), tableName.getSchemaName(), tableName.getTableName()));
+        }
+    }
+
+    /**
+     * Check if identity is allowed to show metadata of tables by executing SHOW TABLES, SHOW GRANTS etc. in a catalog.
+     * <p>
+     * NOTE: This method is only present to give users an error message when listing is not allowed.
+     * The {@link #filterTables} method must filter all results for unauthorized users,
+     * since there are multiple ways to list tables.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanShowTablesMetadata(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, String schemaName)
+    {
+    }
+
+    /**
+     * Check if identity is allowed to insert into the specified table in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanInsertIntoTable(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName)
+    {
+        if (!checkAccess(identity, tableName, null, HiveAccessType.UPDATE)) {
+            denyInsertTable(tableName.toString(), format("Access denied - User [ $s ] does not have is not have [UPDATE] " +
+                    "privilege on [ %s/%s ] ", identity.getUser(), tableName.getSchemaName(), tableName.getTableName()));
+        }
+    }
+
+    /**
+     * Check if identity is allowed to delete from the specified table in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanDeleteFromTable(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName)
+    {
+        if (!checkAccess(identity, tableName, null, HiveAccessType.UPDATE)) {
+            denyDeleteTable(tableName.toString(), format("Access denied - User [ $s ] does not have is not have [UPDATE] " +
+                    "privilege on [ %s/%s ] ", identity.getUser(), tableName.getSchemaName(), tableName.getTableName()));
+        }
+    }
+
+    /**
+     * Check if identity is allowed to create the specified view in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanCreateView(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName viewName)
+    {
+        if (!checkAccess(identity, null, null, HiveAccessType.CREATE)) {
```

Comment:
should we be passing the schema to check for permission within that ?

Reply:
For underline table and column access checks beyond the view individual access callback are triggered to verify.